### PR TITLE
PWN-5099 - Terms and Policy

### PR DESCRIPTION
**Pull request by kuznetsov**
### Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5099

### Description of Work

Open *Terms of Use* and *Privacy Policy* files in onboarding

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/interactor/FileInteractor.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import android.content.Context
+import org.p2p.wallet.auth.repository.FileRepository
+import java.io.File
+
+class FileInteractor(
+    private val context: Context,
+    private val fileRepository: FileRepository
+) {
+
+    companion object {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/FileInteractor.kt:12]: private companion object or extract const outside of class

File: app/src/main/java/org/p2p/wallet/utils/FragmentExtensions.kt [hunk 2]
```diff
@@ -1,11 +1,40 @@
...
+import java.io.File
 import kotlin.reflect.KClass
 
 fun <T : Fragment> Fragment.instantiate(clazz: KClass<T>): Fragment =
     childFragmentManager.fragmentFactory.instantiate(clazz)
 
 fun <T : Fragment> FragmentFactory.instantiate(clazz: KClass<T>): Fragment =
     instantiate(clazz.java.classLoader!!, clazz.java.name)
+
+fun Fragment.openFile(file: File) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/utils/FragmentExtensions.kt:21]: you can put in the BaseFragment as a method

File: app/src/main/res/layout/fragment_new_onboarding.xml [hunk 3]
```diff
@@ -72,13 +73,14 @@
...
         <TextView
+            android:id="@+id/textViewTermsAndPolicy"
             style="@style/UiKit.TextAppearance.Regular.Label1"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:layout_marginBottom="16dp"
             android:gravity="center"
-            android:text="@string/onboarding_terms_button"
-            android:textColor="@color/rain" />
+            android:textColor="@color/rain"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_new_onboarding.xml:82]: text_rain

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/utils/FragmentExtensions.kt (hunk 2 above)
Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/utils/FragmentExtensions.kt:21]: Do you think it's a good idea? It's not so widely used to put it in BaseFragment

File: app/src/main/java/org/p2p/wallet/utils/FragmentExtensions.kt [hunk 4]
```diff
@@ -1,11 +1,40 @@
...
+    val fromFile = FileProvider.getUriForFile(
+        requireContext(),
+        BuildConfig.APPLICATION_ID + ".provider",
+        file
+    )
+
+    val target = Intent(Intent.ACTION_VIEW).apply {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/utils/FragmentExtensions.kt:28]: did you check this method working on Android api 33?

Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/utils/FragmentExtensions.kt:28]: I didn't, just took implementation from previous onboarding

**Pull request merged**